src/coreclr/jit/ssabuilder.cpp

```diff
@@ -1388,58 +1388,67 @@ void SsaBuilder::AddPhiArgsToSuccessors(BasicBlock* block)
 #if defined(TARGET_WASM)
         }
 #endif
-        // Now handle memory.
-        for (MemoryKind memoryKind : allMemoryKinds())
+
+
+#if defined(TARGET_WASM)
+        if (!block->IsLIR())
```

Comment:
I think it would be better to use the approach of `break`ing out of the loop immediately that is used elsewhere:
```scala
        // Now handle memory.
        for (MemoryKind memoryKind : allMemoryKinds())
        {
#ifdef TARGET_WASM
            // TODO-LLVM: LIR memory liveness is NYI upstream. Delete when that is fixed.
            if (block->IsLIR())
            {
                return;
            }
#endif // TARGET_WASM
```
It is less invasive to the code.

Also, there is another `// Now handle memory.` loop below which needs similar treatment.

Reply:
Thanks, will change